### ISSUE TITLE
Provide feedback when no lints are found

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2640]
I think it would be good if `{lintr}` provided some feedback for the success case where there are no lints found instead of returning haunting silence. I am not saying we should go full [`{praise}`](https://cran.r-project.org/web/packages/praise/index.html)-mode, like `{testthat}`. It could simply be a message.

---

``` r
library(withr)
library(lintr)

linters <- all_linters()
``` 

### Failure

``` r
tmp1 <- local_tempfile(lines = "1 + 1", fileext = ".R")
lint(tmp1, linters = linters)
#> /private/var/folders/xr/v_vddzvs33q5wg7jv9mr__4h0000gn/T/RtmpxPFph2/fileca056a1dbb3.R:1:2: style: [implicit_integer_linter] Use 1L or 1.0 to avoid implicit integers.
#> 1 + 1
#> ~^
#> /private/var/folders/xr/v_vddzvs33q5wg7jv9mr__4h0000gn/T/RtmpxPFph2/fileca056a1dbb3.R:1:6: style: [implicit_integer_linter] Use 1L or 1.0 to avoid implicit integers.
#> 1 + 1
#>     ~^
``` 

### Error

```r
tmp2 <- local_tempfile(lines = "1 + ", fileext = ".R")
lint(tmp2, linters = linters)
#> /private/var/folders/xr/v_vddzvs33q5wg7jv9mr__4h0000gn/T/RtmpxPFph2/fileca01d2d9b79.R:1:5: error: [error] unexpected end of input
#> 1 + 
#>     ^
```

### Success

``` r
tmp3 <- local_tempfile(lines = "1L + 1L", fileext = ".R")
lint(tmp3, linters = linters)
```

<sup>Created on 2024-08-02 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

---

Proposed:

``` r
tmp3 <- local_tempfile(lines = "1L + 1L", fileext = ".R")
lint(tmp3, linters = linters)
#> ℹ No lints found!
```
